llvm/utils/TableGen/Common/CodeGenSchedule.h

```diff
@@ -250,6 +250,9 @@ struct CodeGenProcModel {
   // Map from the ReadType field to the parent ReadAdvance record.
   DenseMap<const Record *, const Record *> ReadAdvanceMap;
 
+  // Set of WriteRes that are referenced by a ReadAdvance.
+  DenseSet<const Record *> ReadOfWriteSet;
```

Comment:
nit: should we use SmallPtrSet instead?

from programmers' manual: "...DenseSet is a great way to unique small values that are not simple pointers (use [SmallPtrSet](https://llvm.org/docs/ProgrammersManual.html#dss-smallptrset) for pointers)..."